sktime/forecasting/auto_reg.py

```diff
@@ -0,0 +1,273 @@
+    }
+
+    # todo: add any hyper-parameters and components to constructor
+    def __init__(
+        self,
+        lags=None,
+        trend="c",
+        seasonal=False,
+        hold_back=None,
+        period=None,
+        missing="none",
+        deterministic=None,
+        old_names=False,
+        cov_type="nonrobust",
+        cov_kwds=None,
+        use_t=True,
+        dynamic=False,
+    ):
+        # Model Params
+        self.lags = lags
+        self.trend = trend
+        self.seasonal = seasonal
+        self.hold_back = hold_back
+        self.period = period
+        self.missing = missing
+        self.deterministic = deterministic
+        self.old_names = old_names
+
+        # Fit Params
+        self.cov_type = cov_type
+        self.cov_kwds = cov_kwds
+        self.use_t = use_t
+
+        # Predcit Params
+        self.dynamic = dynamic
+
+        # setup for get_fitted_params
+        self._fitted_param_names = ("aic", "aicc", "bic", "hqic")
+
+        super().__init__()
+
+        # todo: optional, parameter checking logic (if applicable) should happen here
+        # if writes derived values to self, should *not* overwrite self.parama etc
+        # instead, write to self._parama, self._newparam (starting with _)
+
+    # todo: implement this, mandatory
+    def _fit_forecaster(self, y, X=None):
+        """Fit forecaster to training data.
+
+        private _fit_forecaster containing the core logic, called from fit
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        y : guaranteed to be of a type in self.get_tag("y_inner_mtype")
+            Time series to which to fit the forecaster.
+            if self.get_tag("scitype:y")=="univariate":
+                guaranteed to have a single column/variable
+            if self.get_tag("scitype:y")=="multivariate":
+                guaranteed to have 2 or more columns
+            if self.get_tag("scitype:y")=="both": no restrictions apply
+        X : optional (default=None)
+            guaranteed to be of a type in self.get_tag("X_inner_mtype")
+            Exogeneous time series to fit to.
+
+        Returns
+        -------
+        self : reference to self
+        """
+        from statsmodels.tsa.ar_model import AutoReg as _AutoReg
+
+        self._forecaster = _AutoReg(
+            endog=y,
+            lags=self.lags,
+            trend=self.trend,
+            seasonal=self.seasonal,
+            exog=X,
+            hold_back=self.hold_back,
+            period=self.period,
+            missing=self.missing,
+            deterministic=self.deterministic,
+            old_names=self.old_names,
+        )
+
+        self._fitted_forecaster = self._forecaster.fit(
+            cov_type=self.cov_type, cov_kwds=self.cov_kwds, use_t=self.use_t
+        )
+        for param, value in self._fitted_forecaster.params.items():
+            setattr(self, param + "_", value)
```

Comment:
@jonathanbechtel @mgazian000 @CTFallon

This do not seem correct, as it seems to be raising the majority of the failures in CI.

> TypeError: unsupported operand type(s) for +: 'int' and 'str'

For all the failures, `param` seems to be `0`. It seems that feature names in `exog` are few of the parameters, and these can be numeric in absence of column names in `X`.

Reply:
Ok, forced to string in `setattr` and the line beneath it.